winpty/src/windows.rs

```diff
@@ -262,33 +304,26 @@ impl SpawnConfig {
         end: Option<&str>,
     ) -> Result<Self, Err> {
         let mut err = null_mut() as *mut winpty_error_t;
-        let (appname, cmdline, cwd, end) = (
-            appname.map_or(null(), |s| WideCString::from_str(s).unwrap().into_raw()),
-            cmdline.map_or(null(), |s| WideCString::from_str(s).unwrap().into_raw()),
-            cwd.map_or(null(), |s| WideCString::from_str(s).unwrap().into_raw()),
-            end.map_or(null(), |s| WideCString::from_str(s).unwrap().into_raw()),
-        );
+
+        let to_wstring = |s| WideCString::from_str(s).unwrap();
+        let as_ptr = |opt: &Option<WideCString>| opt.as_ref().map_or(null(), |ws| ws.as_ptr());
```

Comment:
These two can be moved closer to the code where they're actually used.

`to_wstring` can be moved right above `let appname =` while the `as_ptr` can be moved right above `spawn_config`.

The `as_ptr` method could also be renamed a bit better. Maybe something more like `wstring_ptr`.

That way we don't accidentally write cobol code where everything's defined at the top and then never used until the bottom of the program is reached.